pkg/client/fake/client.go

```diff
@@ -123,6 +123,12 @@ func (c *fakeClient) Update(ctx context.Context, obj runtime.Object) error {
 	return c.tracker.Update(gvr, obj, accessor.GetNamespace())
 }
 
+func (c *fakeClient) UpdateStatus(ctx context.Context, obj runtime.Object) error {
+	// TODO(droot): This results in full update of the obj (spec + status). Need
+	// a way to update status field only.
+	return c.Update(ctx, obj)
```

Comment:
Couldn't figure out how to implement UpdateStatus behavior with tracker object. 
@mengqiy any ideas ?

Reply:
I don't see an easier way except implementing it upstream.
It seems no one uses fakeclient's UpdateStatus, I think this is OK to be a starting point.

Reply:
Sounds good.